src/index.js

```diff
@@ -1,27 +1,61 @@
 /**
  * @flow
  */
+import * as emails from './store/emails.json';
 
 // example of generating HTML with js
+function renderNewMessage(sender, timestamp, subject, snippet) {
+  const sidebar = document.querySelector('.email-list');
+  const newMessage = document.createElement('li');
+  newMessage.innerHTML = `<button class="email-item" type="button">
+                            <div class="sender-details">
+                              <p>${sender}</p>
+                              <span>${timestamp}</span>
+                            </div>
+                            <p class="email-subject">${subject}</p>
+                            <p>${snippet}</p>
+                          </button>`;
+  sidebar.appendChild(newMessage);
+}
+
+
+function formatTimestamp(datestring) {
+  const d = new Date(datestring);
+  const month = d.getMonth() + 1;
+  const day = d.getDay();
+  const year = String(d.getFullYear()).substring(2);
+  const timestamp = `${month}/${day}/${year}`;
+  return timestamp;
+}
+
+function formatSubject (subject) {
+  return subject.length > 33 ? subject.substring(0, 33) + '...' : subject;
+}
+
+function formatSnippet (snippet) {
+  return snippet.length > 77 ? snippet.substring(0, 77) + '...' : snippet;
+}
+
 function renderSidebar() {
   const sidebarContents = `
     <h2 class="email-header">Inbox</h2>
-    <ul class="email-list">
-      <li>
-        <button class="email-item" type="button">
-          <div class="sender-details">
-            <p>Email sender</p>
-            <span>Timestamp</span>
-          </div>
-          <p class="email-subject">Email1 subject</p>
-          <p>Email snippet</p>
-        </button>
-      </li>
-    </ul>
+    <ul class="email-list"></ul>
   `;
 
   const container = document.querySelector('.email-list-container');
   if (container != null) container.innerHTML = sidebarContents;
+
+  const newMessageIDs = emails.mailboxes.INBOX.threadIds;
+  const allMessages = emails.messages;
+
+  for (const m of newMessageIDs) {
```

Comment:
I would usually create all my HTML in one function (in this case the entire list of emails) rather than appending them one by one to the ul, which is not performant because interacting with the DOM is slow. So in that case, a `map` or `reduce` might be a more readable way to transform your data into the HTML you want and then return the whole HTML as a string. This prevents you from querying the DOM, rendering a string and then appending it for every message and lets you just do this once.

Reply:
@mariehooper Gotcha!